wrappers/preseq/observed_complexity/wrapper.py

```diff
@@ -3,8 +3,20 @@
 __email__ = "behram.radmanesh@nih.gov"
 __license__ = "MIT"
 
+
+import os
 # import snakemake's ability to execute shell commands
 from snakemake.shell import shell
 
+# specify an output directory from snakemake otherwise insert default
+outdir = os.path.dirname(snakemake.output[0])
+if not outdir:
+    outdir = '.'
+
+try:
+    output = snakemake.params.output
+except AttributeError:
+    output = ""
+
 # execute preseq c_curve
-shell("preseq c_curve {snakemake.params} {snakemake.input[0]} {snakemake.output[0]}")
+shell("preseq c_curve {snakemake.params.BAM} {snakemake.input[0]} {snakemake.params.output} {snakemake.output[0]}")
```

Comment:
It looks like here `{snakemake.params.output}` is expected to always be `-o`. Why not just put `-o` here? One less thing to remember to put in the params.